modelling/src/main/java/org/axonframework/modelling/configuration/StateBasedEntityModule.java

```diff
@@ -103,10 +105,25 @@ interface PersisterPhase<I, E> {
          * Registers the given {@code persister} as a factory method for the state-based entity being built.
          *
          * @param persister A factory method constructing a {@link SimpleRepositoryEntityPersister}.
-         * @return The parent {@link StateBasedEntityBuilder}, signaling the end of configuring a state-based entity.
+         * @return The parent {@link StateBasedEntityModule}, signaling the end of configuring a state-based entity.
          */
-        StateBasedEntityBuilder<I, E> persister(
+        EntityModelPhase<I, E> persister(
                 @Nonnull ComponentBuilder<SimpleRepositoryEntityPersister<I, E>> persister
         );
     }
+
+    interface EntityModelPhase<I, E> {
```

Comment:
The `EntityModelPhase` and `EntityIdResolverPhase` should also get some JavaDoc.